packages/phishing-controller/src/types.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Represents the result of checking a domain.
+ */
+export type PhishingDetectorResult = {
+  /**
+   * The name of the configuration object in which the domain was found within
+   * an allowlist, blocklist, or fuzzylist.
+   */
+  name?: string;
+  /**
+   * The version associated with the configuration object in which the domain
+   * was found within an allowlist, blocklist, or fuzzylist.
+   */
+  version?: string;
+  /**
+   * Whether the domain is regarded as allowed (true) or not (false).
+   */
+  result: boolean;
+  /**
+   * A normalized version of the domain, which is only constructed if the domain
+   * is found within a list.
+   */
+  match?: string;
+  /**
+   * Which type of list in which the domain was found.
+   *
+   * - "allowlist" means that the domain was found in the allowlist.
+   * - "blocklist" means that the domain was found in the blocklist.
+   * - "fuzzy" means that the domain was found in the fuzzylist.
+   * - "blacklist" means that the domain was found in a blacklist of a legacy
+   * configuration object.
+   * - "whitelist" means that the domain was found in a whitelist of a legacy
+   * configuration object.
+   * - "all" means that the domain was not found in any list.
+   */
+  type:
+    | 'all'
+    | 'fuzzy'
+    | 'blocklist'
+    | 'allowlist'
+    | 'blacklist'
+    | 'whitelist'
+    | 'c2DomainBlocklist';
+};
```

Comment:
nit: How would you feel about extracting an enum here?

```suggestion
  type: PhishingDetectorResultType;
};

/**
 * The type of list in which the domain was found.
 */
export enum PhishingDetectorResultType {
  /*
   * "all" means that the domain was not found in any list.
   */
  All = 'all',
  /*
   * "fuzzy" means that the domain was found in the fuzzylist.
   */
  Fuzzy = 'fuzzy',
  /*
   * "blocklist" means that the domain was found in the blocklist.
   */
  Blocklist = 'blocklist',
  /*
   * "allowlist" means that the domain was found in the allowlist.
   */
  Allowlist = 'allowlist',
  /*
   * "blacklist" means that the domain was found in a blacklist of a legacy 
   * configuration object.
   */
  Blacklist = 'blacklist',
  /*
   * "whitelist" means that the domain was found in a whitelist of a legacy
   * configuration object.
   */
  Whitelist = 'whitelist',
  /*
   * "c2DomainBlocklist" means that the domain was found in the C2 domain blocklist.
   */
  C2DomainBlocklist = 'c2DomainBlocklist',
}
```